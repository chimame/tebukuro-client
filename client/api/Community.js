import ApiRequest from './ApiRequest'
import request from 'superagent'

export const createCommunityHandler = (name, description) => {
  const communityParams = {
    name: name,
    description: description
  }
  return ApiRequest.post('/communities', communityParams)
}

export const getCommunities = () => {
  return request.get(`${Config.ApiEndPoint}/communities.json`)
    .then(res => res.body)
}
