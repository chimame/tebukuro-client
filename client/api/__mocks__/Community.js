const path = require('path')
const Community = jest.genMockFromModule('../Community')

let mockResult = Object.create(null)
export const __setMockResult = (normal = true) => {
  mockResult = normal
}

export const showCommunity = (id) => mockResult ? Promise.resolve({message: "normal mock"}) : Promise.reject({message: "error mock"})
export const createCommunity = require('../Community').createCommunity
export const getCommunities = require('../Community').getCommunities
