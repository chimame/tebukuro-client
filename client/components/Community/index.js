import React, { Component } from 'react'

export default class Community extends Component {
  render() {
    return (
      <div>
        <p>name: { this.props.community.name }</p>
        <p>description: { this.props.community.description }</p>
      </div>
    )
  }
}

Community.propTypes = {
  community: React.PropTypes.object.isRequired
}
