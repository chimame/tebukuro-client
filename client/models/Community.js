import { Record } from 'immutable'

export const CommunityRecord = Record({
  id:           0,
  name:         '',
  description:  '',
  errors:       []
})

export default class Community extends CommunityRecord {
  isError() {
    return this.errors.length !== 0
  }
}
