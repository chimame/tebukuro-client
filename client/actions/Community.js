import { createAction } from 'redux-actions'
import { routerActions } from 'react-router-redux'
import { createCommunity as createAPI, showCommunity as showAPI } from '../api/Community'

export const CREATE_COMMUNITY = 'CREATE_COMMUNITY'

export const createCommunity = createAction(CREATE_COMMUNITY, createAPI)

export const showCommunity = (id) => {
  return dispatch => {
    return showAPI(id)
      .then(json => dispatch(createCommunity(json)))
      .catch(() => dispatch(routerActions.replace("not_found_url")))
  }
}
