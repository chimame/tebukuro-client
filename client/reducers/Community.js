import { handleActions }  from 'redux-actions'
import Community          from '../models/Community'

export const communityInitialState = new Community()

export const communityReducerMap = {
  CREATE_COMMUNITY: {
    next: (state, action) => {
      return new Community({
        id:           action.payload.id,
        name:         action.payload.name,
        description:  action.payload.description
      })
    },
    throw: (state, action) => {
      return new Community({
        errors: action.payload.messages
      })
    }
  }
}

export default handleActions(communityReducerMap, communityInitialState)
