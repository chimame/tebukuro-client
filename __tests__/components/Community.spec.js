import React          from 'react'
import { shallow }    from 'enzyme'
import shallowToJson  from 'enzyme-to-json'
import CommunityModel from '../../client/models/Community'
import Community      from '../../client/components/Community'
import Params         from '../factories/Community'

const model = (params) => {
  return new CommunityModel(params)
}

describe('Community', () => {
  it('render community component', () => {
    const component = shallow(
      <Community community={ model(Params.community1) } />
    )

    const tree = shallowToJson(component)
    expect(tree).toMatchSnapshot()
  })
})
