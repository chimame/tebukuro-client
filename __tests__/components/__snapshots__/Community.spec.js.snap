// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Community render community component 1`] = `
<div>
  <p>
    name: 
    community1
  </p>
  <p>
    description: 
    This is the first community.
  </p>
</div>
`;
